README.md

```diff
@@ -1 +1,77 @@
-# nvim-package-specification
\ No newline at end of file
+# Plugin metadata specification
+
+The neovim package specification supports a single, top-level package metadata file. This file can be *either* 'plugin.lua' or 'plugin.json'. The file is a superset of the Rockspec File Format and can contain the following fields
+
+* `package` : String, the name of the package
+
+* `version` : String, the version of the package. Should obey semantic versioning conventions, for example `0.1.0`. Plugins should have a git commit with a `tag` matching this version. For all version identifiers, implementation should check for a `version` prefixed with `v` in the git repository, as this is a common convention.
+
+* `specification_version` : String, the current specification version. (0.1.0) at this time.
+
+* `source` : Table, the source is a table that contains a `url` field, which points to the git repository of the current package.
+
+* `description` : Table, the description is a table that includes the following nested fields:
+	* `summary` : String, a short description of the package, typically less than 100 character long.
+	* `detailed` : String, a long-form description of the package, this should convey the package's principal functionality to the user without being as detailed as the package readme.
+	* `homepage` : This is the homepage of the package, which in most cases will be the GitHub URL.
+	* `license` : This is [SPDX](https://spdx.org/licenses/) license identifier. Dual licensing is indicated via joining the relevant licenses via `/`.
+
+* `dependencies`
+** TBD
+
+# Example
+
+```lua
+package = "lspconfig"
+version = "0.1.2"
+specification_version = "0.1.0"
+source = {
+  url = "https://github.com/neovim/nvim-lspconfig.git",
+}
+description = {
+   summary = "Quickstart configurations for the Nvim-lsp client",
+   detailed = [[
+   	lspconfig is a set of configurations for language servers for use with Neovim's built-in language server client. Lspconfig handles configuring, launching, and attaching language servers.
+   ]],
+   homepage = "https://github.com/neovim/nvim-lspconfig/", 
+   license = "Apache-2.0" 
+}
+dependencies = {
```

Comment:
from #1 possible keys suggested by @wbthomason 
```lua
name: String, the canonicalized name of the dependency
reason: Optional string, a short description of the purpose served by the dependency
version: String, a SemVer version constraint specifying the compatible versions of the dependency
type: String, either "plugin" or "rock", specifying if the dependency is a plugin or a Luarocks rock
source: Either string, assumed to be the URL of a git repo, or table, as for the top-level source key.
```

Reply:
Alternatively instead of the "type" field, dependencies could be split into categories:
```lua
dependencies = {
  neovim = { version = "...", source = "..." },
  plugin = {
    ["foo"] = { version = "...", source = "..." },
  },
  rock = {
    ["bar"] = { version = "...", source = "..." },
  },
}
```

Reply:
I'd prefer if we can keep the structure simple and have a `type` field. And tbh something like type can be autodetected by implementations.

Also, I'm not sure about the `reason` field, this can just be a comment. Is there precedence of something like reason in other package formats?

We also need to be mindful of unversioned dependencies with/without a plugin spec.

Reply:
As to unversioned dependencies, the `version` field should just be optional.

I'm thinking about maybe having `tag`, `branch`, `commit` fields for those, but that will most likely often result in conflicting versions.

Reply:
> I'd prefer if we can keep the structure simple and have a type field. And tbh something like type can be autodetected by implementations.

I agree with this

Reply:
Type autodetection seems more expensive than just specifying it here - would that not require resolving a repo to see if it contains a plugin or a rock?

Also re: `reason`: The precedent is package formats like Arch's `PKGBUILD`, which specify optional dependencies with short descriptions explaining why you might want to install the optional dependency. I think this info is helpful but not critical.

Reply:
One other thing to consider is non-git source control, rockspec addresses this by the URI prefix (git:// vs svn:// vs. hg://, etc)